src/parser.rs

```diff
@@ -25,6 +25,9 @@ pub trait Parser<'a, T: Input<'a>> {
     /// Reset a `Parser`.
     fn reset(&mut self);
 
+    /// Returns the queue of all matched `DebugToken`s for debug purposes
+    fn queue_debug(&self) -> Vec<(Self::Token, String)>;
```

Comment:
This method might actually be useful in other cases apart from debugging, so maybe it would be a good idea to name it `queue_with_captures`?